### PR TITLE
Rewrite the badge notification without a timer

### PR DESCRIPTION
**Pull request by daemongh**
To avoid using a timer, the preloaded JS will now be hooking into the `DOMSubtreeModified` event.

Also added some comments, cleaned up some of the code and made it so the notifications should only be sent if there is a change in the count.

### CODE REVIEW

**Comment by lloeki** [comment]
That sounds great, thanks! I'll review this ASAP.

File: src/webview.js [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
+
+document.addEventListener("DOMContentLoaded", function() {
+    // hook into the DOM to detect changes and count the mentions and unread activity
+    $(document).bind("DOMSubtreeModified", function() {
+        checkActivity();
+    });
+
+    // initial one time notification
+    checkActivity();
+});
```
Comment (lloeki) [inline src/webview.js]: No newline at EOF

File: src/webview.js [hunk 2]
```diff
@@ -1,11 +1,45 @@
...
+        unreadCount = localUnreadCount;
+        notify = true;
+    }
+
+    if (notify) notifyHost();
+};
+
+document.addEventListener("DOMContentLoaded", function() {
+    // hook into the DOM to detect changes and count the mentions and unread activity
+    $(document).bind("DOMSubtreeModified", function() {
```
Comment (lloeki) [inline src/webview.js]: Use a [MutationObserver](https://developer.mozilla.org/en/docs/Web/API/MutationObserver) instead.

**Comment by daemongh** [comment]
Okay, it's now using the MutationObserver.

**Comment by lloeki** [comment]
Awesome. Merging.

**Pull request merged**